examples/QuickJournalSync/README.md

```diff
@@ -0,0 +1,386 @@
+- When the application opens, the `LoginPage` is the first to be loaded.
+- If the user is not logged in already (`CurrentUser == null`), then the user can specify its email and password and login. The login method (`RealmService.LoginAsync`) uses `Realm.GetInstanceAsync`, because we want to be sure to have a completely synchronized realm before showing the main page to the user:
+    ```csharp
+    public static async Task LoginAsync(string email, string password)
+    {
+        // ...
+
+        await _app.LogInAsync(Credentials.EmailPassword(email, password));
+
+        // Creates a CancellationTokenSource that will be cancelled after 4 seconds.
+        var cts = new CancellationTokenSource(4000);
+
+        try
+        {
+            using var realm = await Realm.GetInstanceAsync(GetRealmConfig(), cts.Token);
+        }
+        catch (TaskCanceledException)
+        {
+            // If there are connectivity issues, or the synchronization is taking too long we arrive here
+        }
+    }
+    ```
+- If the user just logged in, or was already logged in (`CurrentUser != null`), then we go to the main page (`EntriesPage`). Here, the method to open a realm (`RealmService.GetMainThreadRealm`) uses `Realm.GetInstance`, because we don't need to have the application completely synchronized before showing the main page.
+    ```csharp
+    public static Realm GetMainThreadRealm()
+    {
+        // ...
+        _mainThreadRealm = Realm.GetInstance(mainThreadConfig);
+        // ...
+
+        return _mainThreadRealm;
+    }
+    ```
+
+## Connection state changes
+
+In order to react to connection state changes for the synchronization you need to subscribe to `realm.SyncSession.PropertyChanges` and look out for the `Session.ConnectionState` property, like in the following example:
+
+```csharp
+public static Realm GetRealm()
+{
+    var config = new FlexibleSyncConfiguration(CurrentUser);
+
+    var realm = Realm.GetInstance(config);
+    realm.SyncSession.PropertyChanged += HandleSyncSessionPropertyChanged;
+}
+
+// ...
+
+private static void HandleSyncSessionPropertyChanged(object? sender, PropertyChangedEventArgs e)
+{
+    if (sender is Session session && e.PropertyName == nameof(Session.ConnectionState))
+    {
+        // React to connection state changes
+        LogAndShowToast($"New connection state: {session.ConnectionState}");
+        SyncConnectionStateChanged?.Invoke(null, session.ConnectionState);
+    }
+}
+```
+
+When reacting to connection state changes, please remember that those changes could be momentary, for example due to a drop in the internet connection of the device.
+
+### Simulate connection state changes in the app
+
+You can simulate connection state changes in the app by disabling the internet connection of the testing device, for instance. The app will show a toast when the connection state changes, as well as show the current connection state in the bottom part of the main page.
```

Comment:
Can we put a button to do that? It's kind of difficult to shut down the internet on a simulator, so having a button that does session pause/resume could be nice.

Reply:
Just did it, and modified the README accordingly